src/hasher_impl.rs

```diff
@@ -119,11 +127,10 @@ pub mod blake3 {
         }
 
         fn finalize(&mut self) -> &[u8] {
-            let digest = self.hasher.finalize(); //default is 32 bytes anyway
-            let digest_bytes = digest.as_bytes();
-            let digest_out = &mut self.digest[..digest_bytes.len().max(S)];
-            digest_out.copy_from_slice(digest_bytes);
-            digest_out
+            let Blake3Hasher { hasher, digest } = self;
```

Comment:
yes the types end up being 
```rust
let Blake3Hasher { hasher: &mut Hasher, digest: &mut [u8; S] } = self;
```
i had originally written this wrong and decided to try this out instead of the usual :)
i'll keep change it though since consistency is probably better